obsidiansystems/obelisk
# ob upgrade: v2

# v2

- Switch to using `nix-hash` for hashing
- Verification of graph integrity

## Hashing

The `ob internal hash` command will compute the hash of the current directory. If it is a git repo, untracked (yet to be git-add'ed) files and empty directories will be ignored. It takes an optional revision, in which case it will compute the hash of that specific git tree.

![image](https://user-images.githubusercontent.com/3998/41796468-aca6afcc-7634-11e8-9b50-0d9bb0d64d9d.png)

## Verifying the migration graph

The `ob internal verify-migration` command verifies the integrity of the migration graph:

![image](https://user-images.githubusercontent.com/3998/41796517-de5e0ed4-7634-11e8-83c2-05df9c30aecf.png)


## PR workflow

Before each commit (or at least before the last commit of a PR), we should run `ob internal create-migration` and commit the new migration edge (created between the hash for HEAD and the hash for working copy) along with other changes:

![image](https://user-images.githubusercontent.com/3998/41796342-1ac957a8-7634-11e8-8978-64b656b88b69.png)

## How to test yourself

Install this PR:

```
cd /your/obelisk
git fetch
git checkout ob-upgrade-v2
nix-env -f . -iA command
```

Go to your obelisk project (must be using obelisk `master`), and upgrade it to the PR branch:

```
cd /your/project
ob upgrade ob-upgrade-v2
```

You should see something like:
![image](https://user-images.githubusercontent.com/3998/41796642-58ca62ee-7635-11e8-8bec-6b5dc9efc139.png)
